scripts/fulfillURLManifest.py

```diff
@@ -0,0 +1,110 @@
+        for c in batch['Contents']:
+            currKey = c['Key']
+            metadataObject = s3_client.get_object(Bucket= bucketName, Key= f'{currKey}')
+            update_batch(metadataObject, bucketName, currKey)
+    
+
+def update_batch(metadataObject, bucketName, currKey):
+
+    dbManager = DBManager(
+        user=os.environ.get('POSTGRES_USER', None),
+        pswd=os.environ.get('POSTGRES_PSWD', None),
+        host=os.environ.get('POSTGRES_HOST', None),
+        port=os.environ.get('POSTGRES_PORT', None),
+        db=os.environ.get('POSTGRES_NAME', None)
+    )
+
+    dbManager.generateEngine()
+
+    dbManager.createSession()
+
+    metadataJSON = json.loads(metadataObject['Body'].read().decode("utf-8"))
+    
+    metadataJSON = linkFulfill(metadataJSON, dbManager)
+    metadataJSON = readingOrderFulfill(metadataJSON, dbManager)
+    metadataJSON = resourceFulfill(metadataJSON, dbManager)
+    metadataJSON = tocFulfill(metadataJSON, dbManager)
+
+    fulfillManifest = json.dumps(metadataJSON, ensure_ascii = False, indent = 6)
+
+    try:
+        return s3_client.put_object(Bucket=bucketName, Key=currKey, \
+                            Body=fulfillManifest, ACL= 'public-read', \
+                            ContentType = 'application/json'
+            )
+    except ClientError as e:
+        logging.error(e)
+
+def linkFulfill(metadataJSON, dbManager):
+    for i in metadataJSON['links']:
+        if i['type'] == 'application/pdf' or i['type'] == 'application/epub+zip' \
+            or i['type'] == 'application/epub+xml':
+                for link in dbManager.session.query(Link) \
+                    .filter(Link.url == i['href'].replace('https://', '')):
+                        i['href'] = f'http://127.0.0.1:5050/fulfill/{link.id}'
+                        return metadataJSON
+                
+def readingOrderFulfill(metadataJSON, dbManager):
+    for i in metadataJSON['readingOrder']:
+        if i['type'] == 'application/pdf' or i['type'] == 'application/epub+zip' \
+            or i['type'] == 'application/epub+xml':
+                for link in dbManager.session.query(Link) \
+                    .filter(Link.url == i['href'].replace('https://', '')):
+                        i['href'] = f'http://127.0.0.1:5050/fulfill/{link.id}'
+                        return metadataJSON
+
+def resourceFulfill(metadataJSON, dbManager):
+    for i in metadataJSON['resources']:
+        if i['type'] == 'application/pdf' or i['type'] == 'application/epub+zip' \
+            or i['type'] == 'application/epub+xml':
+                for link in dbManager.session.query(Link) \
+                    .filter(Link.url == i['href'].replace('https://', '')):
+                        i['href'] = f'http://127.0.0.1:5050/fulfill/{link.id}'
+                        return metadataJSON
+
+def tocFulfill(metadataJSON, dbManager): 
+    for i in metadataJSON['toc']:
+        if 'pdf' in i['href'] \
+            or 'epub' in i['href']:
```

Comment:
Is the table of contents section formed differently from the links, resources, etc such that the logic here needs to be different?

Reply:
Yes it is since the toc section does not have a 'type' key value like the links, resources, etc sections. 

"toc": [{"href": "", "title": ""}]